test/external-modules/apm-integration/src/javaRestTest/java/org/elasticsearch/test/apmintegration/MetricsApmIT.java

```diff
@@ -109,7 +108,7 @@ public void testApmIntegration() throws Exception {
         client().performRequest(new Request("GET", "/_use_apm_metrics"));
 
         finished.await(30, TimeUnit.SECONDS);
```

Comment:
I suspect the problem is already here, I'd suggest to make sure this returns true.
```suggestion
        assertTrue("Timeout when waiting for assertions to complete.", finished.await(30, TimeUnit.SECONDS));
```